### PR TITLE
remove unnecessary json parse

### PR DESCRIPTION
**Pull request by Allenskywalker92**
fix failed test case, using id from json result

skip tests that relate with level details

remove present by entity

- [ ] rather 1-story 1-commit than sub-atomic commits

- [ ] commit title is meaningful =>  git history search

- [ ] commit description is helpful => helps the reviewer to understand the changes

- [ ] code is up-to-date with the latest developments of the target branch (rebased to it or whatever) => :fast_forward:-merge for linear history is favoured

- [ ] added code is linted

- [ ] tests are passing (at least locally): we still have some random test failure on CI. thinking of asking spec/examples.txt to be commited

- [ ] in case the changes are visible to the end-user,  video or screenshots should be added to the PR => helps  with user testing

- [ ] testing coverage improvement is improved.

- [ ] CHANGELOG :  add a bullet point on top (optional: reference to github issue/PR )

- [ ] parallele PR for documentation  on docusaurus  if the feature/fix is tagged for a release

### CODE REVIEW

File: app/api/chemotion/molecule_api.rb [hunk 1]
```diff
@@ -188,9 +188,8 @@ class MoleculeAPI < Grape::API
             end
           end
         end
+        
         molecule&.attributes&.merge(temp_svg: svg_process[:svg_file_name], ob_log: ob)
-
-        present molecule, with: Entities::MoleculeEntity
```
Comment (jhmegorei) [inline app/api/chemotion/molecule_api.rb:193]: Why did you remove this line? The change looks rather random... The API should present their objects using Grape Entities. Otherwise you get pretty much undefined results, missing relations defined in MoleculeIdentity etc.

File: spec/features/copy_reaction_spec.rb [hunk 2]
```diff
@@ -81,7 +80,7 @@ def copy_reaction(source_collection, target_collection)
     expect(page).not_to have_button('copy-element-btn', wait: 5)
   end
 
-  it 'to same collection', js: true do
+  xit 'to same collection', js: true do
```
Comment (phuang26) [inline spec/features/copy_reaction_spec.rb:83]: why?

File: app/packs/src/components/fetchers/SamplesFetcher.js [hunk 3]
```diff
@@ -82,8 +82,8 @@ export default class SamplesFetcher {
       },
       body: JSON.stringify(sample.serialize())
     }).then(response => response.json())
-      .then(json => GenericElsFetcher.uploadGenericFiles(sample, json.sample.id, 'Sample')
-        .then(() => this.fetchById(json.sample.id))).catch((errorMessage) => {
+      .then(json => GenericElsFetcher.uploadGenericFiles(json, json.id, 'Sample')
```
Comment (phuang26) [inline app/packs/src/components/fetchers/SamplesFetcher.js:85]: why?

File: app/api/chemotion/molecule_api.rb (hunk 1 above)
Comment (jhmegorei) [reply app/api/chemotion/molecule_api.rb:193]: I see. But then the correct move would be to add these additional attributes to the entity (using the context hash allows this, see ApplicationEntity#displayed_in_list? for how to implement it), not removing the entity entirely.

Returning the attributes hash directly is undefined behaviour as new attributes are added or old ones deleted over time. Using the entity, the returned data structure can be inferred directly without looking up the current data model of the molecule table.

File: app/packs/src/components/fetchers/SamplesFetcher.js (hunk 3 above)
Comment (Allenskywalker92) [reply app/packs/src/components/fetchers/SamplesFetcher.js:85]: @phuang26 I will revert it.

File: app/api/chemotion/molecule_api.rb (hunk 1 above)
Comment (Allenskywalker92) [reply app/api/chemotion/molecule_api.rb:193]: I updated it like you said. Please help me to take a loook.

Comment (jhmegorei) [reply app/api/chemotion/molecule_api.rb:193]: This should work fine :) The options hash access evaluates to nil so any other place using the MoleculeEntity will always get nil. While it is possible to only render the temp_svg and ob_log (what is an ob_log anyway? could we refactor this variable here and in all other related places to have a meaningful name?) only when not nil, I don't consider that strictly necessary.

Comment (Allenskywalker92) [reply app/api/chemotion/molecule_api.rb:193]: While it is possible to only render the temp_svg and ob_log - Could you show me how to achieve it?
What is an ob_log anyway? could we refactor this variable here and in all other related places to have a meaningful name? - Actually I don't know either, I fix here for it to run correctly at first, we can improve it later.

Comment (jhmegorei) [reply app/api/chemotion/molecule_api.rb:193]: The MoleculeEntity actually contains an example of how it works :) see the separate exposure of  :molecule_names ? the :unless parameter receives a Proc with the options as parameter, so you could write it like this:

`expose :temp_svg, unless: ->(instance, options) { options[:temp_svg].present? }`

File: spec/features/copy_reaction_spec.rb (hunk 2 above)
Comment (Allenskywalker92) [reply spec/features/copy_reaction_spec.rb]: updated

File: app/api/chemotion/molecule_api.rb (hunk 1 above)
Comment (Allenskywalker92) [reply app/api/chemotion/molecule_api.rb:193]: updated

Comment (jhmegorei) [reply app/api/chemotion/molecule_api.rb:193]: 👍

**Review by PiTrem — CHANGES_REQUESTED** [review]
check report feature is working

**Comment by Allenskywalker92** [comment]
> check report feature is working

@PiTrem The report only works after my changes.

**Review by PiTrem — APPROVED** [review]
